LibSWBF2.NET/Utils/MarshalingUtils.cs

```diff
@@ -28,6 +48,32 @@ public static T[] IntPtrToWrapperArray<T>(IntPtr nativePtr, int count) where T :
             return wrappers;
         }
 
+
+        // Using the memcpy wrapper (Memory_Blit), this method will extract an array of unmanaged
+        // types from a native buffer. Works for all primitive unmanaged types, as well as unmanaged structs
+        // (e.g. VertexWeight and soon the Vector types).
+        public static T[] IntPtrToArray<T>(IntPtr dataPtr, int count) where T : unmanaged
+        {
+            T[] array = new T[count];
+
+            if (dataPtr == IntPtr.Zero || count == 0) return array;
+            
+            unsafe
+            {
+                int numBytes = count * sizeof(T);
+                T* srcPtr = (T*) dataPtr.ToPointer();
+                fixed (T* destPtr = array)
+                {
+                    APIWrapper.Memory_Blit((void *) destPtr, (void *) srcPtr, numBytes);
+                }
+            }    
+            
+            return array;
+        }
+
+
```

Comment:
Ensuring that Vector2/3 cpp structs are packed and aligned lets me marshal Vector arrays cleanly.  Not tested on Windows.

Reply:
Alignment / packaging can be such a pain in the ass when marshalling. I removed the inheritance hierarchy in the vectors last tweek or so, so there're no more virtual tables attached that mess around.